### PR TITLE
Add documentation about the MPI bind model

### PR DESCRIPTION
**Pull request by gvallee**
Signed-off-by: Geoffroy Vallee <geoffroy.vallee@gmail.com>

## Description of the Pull Request (PR):

Add the documentation related to MPI and the bind model.

## This fixes or addresses the following GitHub issues:

- Fixes #

### CODE REVIEW

File: mpi.rst [hunk 1]
```diff
@@ -192,12 +199,86 @@ If the host MPI is Open MPI, the definition file looks like:
       echo "Compiling the MPI application..."
       cd /opt && mpicc -o mpitest mpitest.c
 
-The standard way to execute MPI applications with Singularity containers is to
+Bind model
+----------
+
+Similarily to the *Hybrid Approach*, the basic idea behind *Bind Approach* is to start the MPI
+application by calling the MPI launcher (e.g., `mpirun`) from the host. The main different between
```
Comment (dtrudg) [inline mpi.rst]: different -> difference

File: mpi.rst [hunk 2]
```diff
@@ -192,12 +199,86 @@ If the host MPI is Open MPI, the definition file looks like:
...
+    the version of MPI available on the host.
+  - The user must know where the host MPI is installed.
+  - The user must ensure that binding the directory where the host MPI is installed is
+    possible.
+  - The user must ensure that the host MPI is compatible with the MPI used to compile
+    and install the application in the container.
+
+The creation of a Singularity container based on the bind model is based on the following steps:
+
+1. Compile your application on a system with the target MPI implementation, as you would docker
```
Comment (dtrudg) [inline mpi.rst]: docker -  what's the docker reference here?

**Review by dtrudg — CHANGES_REQUESTED** [review]
Very minor details.

File: mpi.rst (hunk 2 above)
Comment (gvallee) [reply mpi.rst]: I have no idea, I suspect auto-correct. Will fix.

**Review by dtrudg — APPROVED** [review]

**Pull request merged**